### PR TITLE
Fix SQL expr & type array bounds parsing

### PR DESCRIPTION
**Pull request by aljazerzen**

### CODE REVIEW

File: edb/common/ast/codegen.py [hunk 1]
```diff
@@ -52,6 +52,15 @@ def node_visit(self, node: base.AST) -> None:
...
 
+    def visit_indented(self, node: base.AST, indent: bool=False) -> None:
+        if indent:
+            self.new_lines = 1
+            self.char_indentation += 1
+        res = self.visit(node)
+        if indent:
+            self.char_indentation -= 1
+        return res
+
```
Comment (aljazerzen) [inline edb/common/ast/codegen.py:63]: I've added this helper that could probably simplify pg codegen a lot. If accepted here, I may do another refactor on it.

Comment (msullivan) [reply edb/common/ast/codegen.py:63]: Yeah that looks good. Should `indent` default to `True` here, though?

File: edb/common/ast/codegen.py [hunk 2]
```diff
@@ -52,6 +52,22 @@ def node_visit(self, node: base.AST) -> None:
         visitor = getattr(self, method, self.generic_visit)
         return visitor(node)
 
+    def visit_indented(
+        self, node: base.AST, indent: bool = False, nest: bool = False
```
Comment (msullivan) [inline edb/common/ast/codegen.py:56]: Should indent be true by default?

File: edb/pgsql/ast.py [hunk 3]
```diff
@@ -884,22 +878,21 @@ def copyfrom(self, other):
...
 class SubLinkType(enum.IntEnum):
-    EXISTS = enum.auto()
-    NOT_EXISTS = enum.auto()
-    ALL = enum.auto()
-    ANY = enum.auto()
-    EXPR = enum.auto()
+    pass
+    pass
+    pass
+    pass
```
Comment (msullivan) [inline edb/pgsql/ast.py:884]: Maybe just one `pass`.

Comment (msullivan) [reply edb/pgsql/ast.py:884]: Wait, no, maybe it can just go away :)

**Review by msullivan — APPROVED** [review]
Great cleanup

File: edb/common/ast/codegen.py (hunk 2 above)
Comment (aljazerzen) [reply edb/common/ast/codegen.py:56]: Ah, yes. I wanted to just override `visit`, but typing complained extensively.

File: edb/pgsql/ast.py (hunk 3 above)
Comment (aljazerzen) [reply edb/pgsql/ast.py:884]: Maybe I should start re-reading my commits before pushing :D

**Pull request merged**